app/code/Magento/AdvancedSearch/etc/adminhtml/system.xml

```diff
@@ -40,27 +42,28 @@
                        sortOrder="82"
                        showInDefault="1"
                        showInWebsite="1"
-                       showInStore="1">
+                       showInStore="1"
+                       canRestore="1">
                     <label>Show Results Count for Each Recommendation</label>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                     <depends>
                         <field id="search_recommendations_enabled">1</field>
                     </depends>
                 </field>
                 <!--<group id="suggestions">-->
-                    <field id="search_suggestion_enabled" translate="label comment" type="select" sortOrder="90" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <field id="search_suggestion_enabled" translate="label comment" type="select" sortOrder="90" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
```

Comment:
Nice catch with `canRestore` as default value is available in `config.xml` file

https://github.com/magento/magento2/blob/b75b478d20827acd57494dc1ebc3bea40d0de3e7/app/code/Magento/AdvancedSearch/etc/config.xml#L12